notebooks/integrations/gemini/vector-search-gemini-elastic.ipynb

```diff
@@ -0,0 +1,206 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "c52e30d1-cb29-4e70-af4a-9c953fcb0f2e",
+   "metadata": {},
+   "source": [
+    "# Quickstart: Vector search using Gemini Embeddings and Elasticsearch\n",
+    "\n",
+    "This tutorial demonstrates how to use the [Gemini API](https://ai.google.dev/docs) to create [embeddings](https://ai.google.dev/docs/embeddings_guide) and store them in Elasticsearch. Elasticsearch will enable us to perform vector search (Knn) to find similar documents."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88303061-f357-43d8-8b63-c4f79e9a1746",
+   "metadata": {},
+   "source": [
+    "## setup\n",
+    "\n",
+    "* Elastic Credentials - Create [Cloud deployment](https://www.elastic.co/search-labs/tutorials/install-elasticsearch/elastic-cloud) to get all Elastic credentials (`elastic_user`, `elastic_password`, `cloud_id`).\n",
+    "\n",
+    "* `GOOGLE_API_KEY` - To use the Gemini API, you need an API key. [Follow](https://ai.google.dev/tutorials/setup) to create a key with one click in Google AI Studio."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "76ca723c-6148-4682-a5ae-486e73cb2b94",
+   "metadata": {},
+   "source": [
+    "## Install packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ef1f1e52-f892-489f-8947-3e4698f5f5c3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pip install -q -U google-generativeai elasticsearch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3d86d3fa-4ca0-41b6-a4bc-81bacf26bf02",
+   "metadata": {},
+   "source": [
+    "## Import packages and credentials"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bb62d8fb-6c34-44fd-bc94-18b644422ee8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import google.generativeai as genai\n",
+    "import google.ai.generativelanguage as glm\n",
+    "from elasticsearch import Elasticsearch, helpers\n",
+    "from dotenv import load_dotenv\n",
+    "import os, json\n",
+    "\n",
+    "load_dotenv()\n",
+    "\n",
+    "google_api_key=os.getenv('GOOGLE_API_KEY')\n",
+    "elastic_cloud_id=os.getenv('ES_CLOUD_ID')\n",
+    "elastic_user=os.getenv('ES_USER')\n",
+    "elastic_password=os.getenv('ES_PASSWORD')\n",
```

Comment:
I believe all (or at least most) of our notebooks use the `getpass()` function to get credentials, since setting environment variables is not as easy from a notebook. We also prefer using API Keys and not the Elastic password.

Would it be possible to make these changes here as well? Have a look at [00-quick-start.ipynb](https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/00-quick-start.ipynb) for an example.

Reply:
Hi Miguel, Thanks for Suggestion. Notebook updated accordingly.